### PR TITLE
Use Blah(*) to optimise data sent

### PR DESCRIPTION
**Pull request by brian-brazil**
Still have to write unittests for this.

### CODE REVIEW

**Comment by frank-weinberg** [comment]
I still think we should move to matching whole path components instead of character by character.

**Comment by brian-brazil** [comment]
That'd take a lot of checking of the current code, a lot depends on
prefixes currently. The CPU cost is almost not significant.

**Comment by brian-brazil** [comment]
Unittests added.

File: tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        assertTrue(pt.covers("ScoreBoard.PeriodFoo"));
+
+        assertFalse(pt.covers("ScoreBoard.Perioc"));
+        assertFalse(pt.covers("ScoreBoard.Perioe"));
+        assertFalse(pt.covers("ScoreBoard(1).Perioe"));
+        assertFalse(pt.covers("ScoreBoard.Perio"));
+        assertFalse(pt.covers("Scoreboard.Period"));
+    }
+
+    public void path_trie_overlap() {
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java:34]: All but the first of your tests are missing the `@Test` annotation and are thus not run.

File: tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        assertTrue(pt.covers("ScoreBoard.Period(1)"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).StarPass"));
+    }
+
+    public void path_trie_id_glob() {
+        pt.add("ScoreBoard.Period(*).Jam(1).Foo(*).Bar");
+
+        assertTrue(pt.covers("ScoreBoard.Period(1)"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2).Bar"));
+        assertTrue(pt.covers("ScoreBoard.Period(1).Jam(1).Foo(2).Bar.Baz"));
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java:60]: This fails (and it should).

File: tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+        assertTrue(pt.covers("ScoreBoard.Period*"));
+        assertTrue(pt.covers("ScoreBoard.Period*a"));
+
+        assertFalse(pt.covers("ScoreBoard.Period"));
+        assertFalse(pt.covers("ScoreBoard.Period("));
+        assertFalse(pt.covers("ScoreBoard.Period(1).Jam(2).Foo(2).Bar"));
+    }
+
+}
```
Comment (frank-weinberg) [inline tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java:83]: Maybe add a test where you combine `Period(1).Foo` and `Period(*).Bar`.

File: html/controls/operator.js [hunk 4]
```diff
@@ -1166,8 +1170,6 @@ function createTimeoutDialog() {
 			firstJamListed[i] = 0;
 			lastJamListed[i] = 0;
 			jamDropdownTemplate[i] = $('<select>').attr('id', 'JamDropdown').attr('period', i);
-			WS.Register(['ScoreBoard.Period('+i+').CurrentJamNumber', 'ScoreBoard.Period('+i+').FirstJamNumber'], processJamNumber);
-			WS.Register(['ScoreBoard.Period('+i+').Timeout'], processTimeout);
 			lastPeriodRegistered = i;
```
Comment (frank-weinberg) [inline html/controls/operator.js]: This creates a race. If for a completed period both calls to `processJamNumber` are made before this code is run, the period will not be initialised properly.

**Review by frank-weinberg — COMMENTED** [review]
So registering a ton of unnecessary elements does have a notable performance impact after all.

**Comment by brian-brazil** [comment]
Updated.

File: html/controls/operator.js (hunk 4 above)
Comment (brian-brazil) [reply html/controls/operator.js]: I've reworked this.

File: tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java (hunk 3 above)
Comment (frank-weinberg) [reply tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java:83]: Did you miss this one?

**Comment by frank-weinberg** [comment]
As I mentioned on an earlier PR: Force pushing your changes into the existing commits means I effectively have to review the entire PR again. Please don't do that on nontrivial PRs.

File: tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java (hunk 3 above)
Comment (brian-brazil) [reply tests/com/carolinarollergirls/scoreboard/jetty/WSTest.java:83]: path_trie_overlap_glob does this now.

**Comment by brian-brazil** [comment]
If you click force-pushed, you can see the diff. I like the ultimate commit history being clean.

**Pull request merged**